File: frontend/src/types/camera.ts
import type { KpiDto, PointDto } from './common'

export type CameraStatus = 'Hoạt động' | 'Lỗi' | 'Mất tín hiệu'

export interface CameraItem {
  id: string
  code: string
  name: string
  room: string
  ipAddress: string
  brand: string
  model: string
  resolution: string
  status: CameraStatus
  installedAt: string
  warranty: string
  note: string
}

export interface CameraManagementResponse {
  overview: KpiDto[]
  statusData: PointDto[]
  roomLoad: PointDto[]
  items: CameraItem[]
  statuses: string[]
}

File: backend/Data/Entities/Camera.cs
namespace backend.Data.Entities;

public class Camera
{
    public int Id { get; set; }
    public string Code { get; set; } = "";
    public string Name { get; set; } = "";
    public int RoomId { get; set; }
    public string IpAddress { get; set; } = "";
    public string Brand { get; set; } = "";
    public string Model { get; set; } = "";
    public string Resolution { get; set; } = "";
    public string Status { get; set; } = "";
    public string InstalledAt { get; set; } = "";
    public string Warranty { get; set; } = "";
    public string Note { get; set; } = "";

    // Navigation properties
    public Room? Room { get; set; }
}
